src/content/platform-integration/web/embedding-flutter-web.md

```diff
@@ -0,0 +1,326 @@
+method, as shown:
+
+```js highlightLines=4-7
+// Adding a view with initial data...
+let viewId = app.addView({
+  hostElement: someElement,
+  initialData: {
+    greeting: 'Hello, world!',
+    randomValue: Math.floor(Math.random() * 100),
+  }
+});
+```
+
+In Dart, the `initialData` is available as a `JSAny` object, accessible through
+the top-level `views` property in the `dart:ui_web` library. The data is
+accessed through the `viewId` of the current view,  as shown:
+
+```dart
+final initialData = ui_web.views.getInitialData(viewId) as YourJsInteropType;
+```
+
+To learn how to define the `YourJsInteropType` class to map the `initialData`
+object passed from JS so it's type-safe in your Dart program, check out:
+[JS Interoperability][] on dart.dev.
+
+[JS Interoperability]: {{site.dart-site}}/interop/js-interop
+
+### View constraints
+
+By default, an embedded Flutter web view considers the size of its `hostElement`
+as an immutable property, and tightly constrains its layout to the available
+space.
+
+On the web, it's common for the intrinsic size of an element to affect the
+layout of the page (like `img` or `p` tags that can reflow content around
+them).
+
+When adding a view to Flutter web, you may configure it with constraints that
+inform Flutter of how the view needs to be laid out:
+
+```js highlightLines=4-8
+// Adding a view with initial data...
+let viewId = app.addView({
+  hostElement: someElement,
+  viewConstraints: {
+    maxWidth: 320,
+    minHeight: 0,
+    maxHeight: Infinity,
+  }
+});
+```
+
+The view constraints passed from JS need to be compatible with the CSS styling
+of the `hostElement` where Flutter is being embedded. For example, Flutter
+won’t try to "fix" contradictory constants like passing  `max-height: 100px`
+in CSS, but `maxHeight: Infinity` to Flutter.
+
+To learn more, check out the [`ViewConstraints` class][],
+and [Understanding constraints][].
+
+[`ViewConstraints` class]: {{site.api}}/flutter/dart-ui/ViewConstraints-class.html
+[Understanding constraints]: {{site.docs}}/ui/layout/constraints
+
+## Custom element (`hostElement`)
+
+_Between Flutter 3.10 and 3.24_<br />
+You can embed a single-view Flutter web app into any HTML element of your web
+page.
+
+To tell Flutter web in which element to render, pass an object with a `config`
+field to the `_flutter.loader.load` function that specifies a `HTMLElement` as
+the `hostElement`.
+
+```js highlightLines=3
+_flutter.loader.load({
+  config: {
+    hostElement: document.getElementById('flutter_host'),
+  }
+});
+```
+
+To learn more about other configuration options,
+check out [Customizing web app initialization][].
+
+:::version-note
+This method of specifying the `hostElement` is superseded by the
+**Embedded mode** described above, **please consider migrating to it**.
+To learn how to configure the `hostElement` in earlier Flutter versions,
+reference [Legacy web app initialization][].
+:::
```

Comment:
```suggestion
#### Flutter 3.24 and beyond

As of the 3.24 release, specifying a custom element by
using **Embedded mode** (described above),
supersedes specifying the `hostElement` directly.
Please migrate to the Embedded mode approach.

To learn how to configure the `hostElement` in Flutter versions before 3.24,
check out [Legacy web app initialization][].
```

Reply:
> why not document embedded mode _first_ and then mention hostname second.

Yep, that's exactly what we're doing; check it out here:

https://flutter-docs-prod--pr10977-web-embedding-docs-o3h28ltv.web.app/platform-integration/web/embedding-flutter-web#embedded-mode

Reply:
D'OH!